### ISSUE TITLE
kernel-panic by kernel 4.14 when linuximq should start.

### ISSUE THREAD

**Issue by Roland-F** [imq/linuximq#68]
Hello,
with the kernel 4.14 i will get during the startup of my system an kernel-panic (As already told you for the kernel 4.14 series -> https://github.com/imq/linuximq/issues/66)
I figured now out the exactly point, when the fault comes up.
The commands will be:
tc qdisc add dev imq0 root handle 10: htb default 6 r2q 1
tc class add dev imq0 parent 10: classid 10:7 htb rate 50000kbit 
tc qdisc add dev imq0 parent 10:7 sfq  perturb 10
tc qdisc add dev imq1 root handle 20: htb default 4 r2q 1 
tc class add dev imq1 parent 20: classid 20:7 htb rate 10000kbit 
tc qdisc add dev imq1 parent 20:7 sfq  perturb 10
tc class add dev imq1 parent 20:7 classid 20:1 htb rate 200kbit ceil 10000kbit  prio 0
(Normaly there will follow some more lines from me configuration)

and exactly when the last line will be activated, the kernel-panic appears.
I use now also iproute2 in version 4.14.1 together with an kernel 4.14.14.

Has someone an identical problem and has maybe solved this already?
It will be great if there come some hints to me, how to solve this.

**Comment by k0ste** [comment]
Try to use your physdev instead imq device.

```shell
[root@linux01 ~]# tc qdisc add dev eth0 root handle 20: htb default 4 r2q 1
[root@linux01 ~]# tc class add dev eth0 parent 20: classid 20:7 htb rate 10000kbit
[root@linux01 ~]# tc qdisc add dev eth0 parent 20:7 sfq perturb 10
[root@linux01 ~]# tc class add dev eth0 parent 20:7 classid 20:1 htb rate 200kbit ceil 10000kbit prio 0
Segmentation fault (core dumped)
```

You will see the same. It's a kernel bug.

```
[Sat Jan 20 12:43:06 2018] BUG: scheduling while atomic: tc/4421/0x00000202
```

Please report to `netdev` about this issue.

**Comment by Roland-F** [comment]
Fault reported in NETDEV group and give the information, that the fault is also in Kernel 4.15 (4.15.1) inside.
Ask them also to solve that problem including the backport for the 4.14 kernel-line.
Hope that the fix come with the next version (4.15.2 and 4.14.18).
When the next version is out and fix the problem (Kernel 4.14 and 4.15) for me, i will report this here.

**Comment by Roland-F** [comment]
Function will not working with 4.15.2 and 4.14.18 as patch for that is not include.
Pre-Patch for the Kernel 4.15 is on: https://patchwork.ozlabs.org/patch/844372/
This patch will work by me and Linuximq can be activated suessfull.
For Kernel 4.14 i have no address as between 4.14 and 4.15 are a lot of things be changed.

**Comment by Roland-F** [comment]
Kernel 4.15.3 has the fix included. This will working also for me (No seperate patch need).

**Comment by Roland-F** [comment]
Not solved in kernel 4.14.20.
Still open there.

**Comment by Roland-F** [comment]
Today is the need patch updated in Kernel 4.14 stable. So the next version (4.14.24) should fix the problem.

**Comment by Roland-F** [comment]
Fixed with kernel 4.14.24 and higher.
